invenio_records_resources/services/records/service.py

```diff
@@ -651,13 +651,17 @@ def _create_record(record_dict):
                 _create_record(record_dict)
             except Exception as exc:
                 records_processed.append(("create", record_dict, None, exc))
-
-        # We only commit records that have no errors
-        records = [
-            record
-            for _, record, errors, exc in records_processed
-            if errors == [] and exc is None
-        ]
-        uow.register(RecordBulkCommitOp(records, self.indexer))
+        valid_records = []
+
+        for i, (action, record, errors, exc_) in enumerate(records_processed):
+            if errors == [] and exc_ is None:
+                try:
+                    record.commit()
+                    valid_records.append(record)
+                except Exception as exc:
```

Comment:
Can we have a more specific exception here instead?

Reply:
I added the base `RelationError`, to my understanding all validation errors happening on commit for the system field should inherit from this. This works at the moment, let me know if I also should add the general Exception or just keep it like this.